src/stream/src/executor/mview/materialize.rs

```diff
@@ -149,11 +158,15 @@ impl<S: StateStore, SD: ValueRowSerde> MaterializeExecutor<S, SD> {
                         .with_label_values(&[&table_id_str, &actor_id_str, &fragment_id_str])
                         .inc_by(chunk.cardinality() as u64);
 
+                    // This is an optimization that handles conflicts only when a particular materialized view downstream has no MV dependencies.
+                    // This optimization is applied only when there is no specified version column and the is_consistent_op flag of the state table is false.
+                    let do_not_handle_conflict =
```

Comment:
I think we still need to handle conflict for `ConflictBehavior::Overwrite` and `ConflictBehavior::DoUpdateIfNotNull` even when there is no version column. because it will change the result stored in the storage